src/web-mercator-utils.js

```diff
@@ -266,3 +278,69 @@ export function getProjectionMatrix({
 
   return projectionMatrix;
 }
+
+/**
+ * mapbox-gl-js flyTo : https://www.mapbox.com/mapbox-gl-js/api/#map#flyto.
+ * It implements “Smooth and efficient zooming and panning.” algorithm by
+ * "Jarke J. van Wijk and Wim A.A. Nuij"
+*/
+/* eslint-disable max-statements */
+export function flyToViewport(startProps, endProps, t) {
```

Comment:
Nit: Feels like it should have a separate file. This function stands out as pretty special.